src/main/java/seedu/ecardnomics/game/GameStorage.java

```diff
@@ -0,0 +1,34 @@
+package seedu.ecardnomics.game;
+
+import seedu.ecardnomics.deck.Deck;
+import seedu.ecardnomics.deck.FlashCard;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class GameStorage {
+    Deck originalDeck;
+    ArrayDeque<FlashCard> deque;
+    ArrayList<FlashCard> retestStore;
+
+    GameStorage(Deck deck) {
+        originalDeck = deck;
+        deque = createRandomisedStack(deck);
+        retestStore = new ArrayList<>();
+    }
+
+    ArrayDeque<FlashCard> createRandomisedStack(Deck deck) {
+        Collections.shuffle(deck.getDeck());
+        return new ArrayDeque<>(deck.getDeck());
```

Comment:
Should we copy into the new ArrayDeque before shuffling? Do we want to keep the original order of the Deck?

Reply:
Oh this one the idea is to make a stack of questions in jumbled order - that's why first shuffle then make a stack out of it. So it's like pre-determined order, but to the user it will be "random". Implementation-wise, it's just popping off the topmost question from the "jumbled" stack. Idk if this answers your question?

Reply:
Sorry the earlier comment wasn't very clear. What I meant was that since we have

`ArrayList<FlashCard> getDeck() {
    return deck;
}`

then would passing the reference to the ArrayList to shuffle cause the original deck to be randomized also?

If we want  to maintain the order in the original deck then perhaps a suggestion could be changing getDeck() in Deck class to

`ArrayList<FlashCard> getDeck() {
    return new ArrayList<>(deck);
}`

so that other methods cannot modify the ArrayList<FlashCards> inside the Deck object except through add and delete.

Reply:
I misunderstood this question! I have since fixed the issue. Thanks!